lzyang2000/Avia-X
# create user management system and basic preset themes

This PR adds a mock interactive agent and a mock pickle-based database to
implement the most basic functions of the user system. The system is
refactor-ready and easy to integrate.

# How to Test
Run `launch.py` with python3, and simply follow the instructions. By the end of the session, you should be able to create a new user.
Run `launch.py` again, and you should be able to see your new user. You could login by typing the username. This mimics Netflix's login UX: you tap your avatar on the screen to log in.
Script `/user/clear_user_database.py` clears all user data.

# Notes
The user database is deliberately not added to `.gitignore` so we could keep a few users in shared database, which is good for developing learning algorithm in the future.
The function of the `Agent` class may seem confusing. `Agent` is the user's interactive device, which is the console in this case.

# What's next
The real system should host at most 4 `Agent` instances in parallel. Running 4 `mock_agent` instances is achievable but inelegant in implementation. We should talk about where to go next from here.
